components/ResearchCoin/ClaimRewardsModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { StyleSheet, css } from "aphrodite";
+import BaseModal from "../Modals/BaseModal";
+import colors from "~/config/themes/colors";
+import { useState } from "react";
+import ProgressStepper, {
+  ProgressStepperStep,
+} from "../shared/ProgressStepper";
+import Button from "../Form/Button";
+import IconButton from "~/components/Icons/IconButton";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faArrowLeft,
+  faInfoCircle as faInfoCircleLight,
+} from "@fortawesome/pro-light-svg-icons";
+import { submitRewardsClaim } from "./lib/api";
+import useCurrentUser from "~/config/hooks/useCurrentUser";
+import { ID } from "~/config/types/root_types";
+
+interface Props {
+  isOpen: boolean;
+  paperId: ID;
+  closeModal: () => void;
+}
+
+export type STEP =
+  | "INTRO"
+  | "OPEN_ACCESS"
+  | "OPEN_DATA"
+  | "PREREGRISTRATION"
+  | "CLAIM_SUBMITTED";
+
+export const ORDERED_STEPS: STEP[] = [
+  "INTRO",
+  "OPEN_ACCESS",
+  "OPEN_DATA",
+  "PREREGRISTRATION",
+  "CLAIM_SUBMITTED",
+];
+
+const stepperSteps: ProgressStepperStep[] = [
+  {
+    title: "Open Access",
+    number: 1,
+    value: "OPEN_ACCESS",
+  },
+  {
+    title: "Open Data",
+    number: 2,
+    value: "OPEN_DATA",
+  },
+  {
+    title: "Preregistration",
+    number: 3,
+    value: "PREREGRISTRATION",
+  },
+];
+
+const ClaimRewardsModal = ({ paperId, isOpen, closeModal }: Props) => {
+  const [step, setStep] = useState<STEP>("INTRO");
+  const currentUser = useCurrentUser()
+
+  const handleSubmitClaim = () => {
+
+    if (!currentUser) {
+      return;
+    }
+
+    // @ts-ignore Temporarily ignoring due to hard-coding 
+    submitRewardsClaim({
+      paperId,
+      authorshipId: 1762,
```

Comment:
@koutst for testing you will need to include an id for any record from `paper_authorship` table

Reply:
Nit: Maybe add a `// FIXME` here to make it easier to revisit this?